matter_server/server/server.py

```diff
@@ -219,11 +220,13 @@ def get_diagnostics(self) -> ServerDiagnostics:
 
     def signal_event(self, evt: EventType, data: Any = None) -> None:
         """Signal event to listeners."""
+        if TYPE_CHECKING:
+            assert self.loop
         for callback in self._subscribers:
             if asyncio.iscoroutinefunction(callback):
                 asyncio.create_task(callback(evt, data))
             else:
-                callback(evt, data)
+                self.loop.call_soon_threadsafe(callback, evt, data)
```

Comment:
Hm, I guess technically this is not needed in every `signal_event` call but maybe better safe than sorry :thinking: